cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ApiCredentials.java

```diff
@@ -0,0 +1,225 @@
+     */
+    public record Entry(String username, String password, Role role) { }
+
+    /**
+     * Parse String containing API credential config into map of API user entries.
+     *
+     * @param config API credential config file as a String.
+     *
+     * @return map of API credential entries containing username, password, and role.
+     */
+    public static Map<String, Entry> parseEntriesFromString(String config) {
+        Map<String, Entry> entries = new HashMap<>();
+        for (String line : config.split("\n")) {
+            Matcher matcher = HASH_LOGIN_SERVICE_PATTERN.matcher(line);
+            if (matcher.matches()) {
+                String[] parts = line.replaceAll("\\s", "").split(":");
+                String username = parts[0];
+                String password = parts[1].split(",")[0];
+                Role role = Role.fromString(parts[1].split(",")[1]);
+                if (entries.containsKey(username)) {
+                    throw new InvalidConfigurationException("Duplicate username found: " + "\"" + username + "\". "
+                            + "Cruise Control API credentials config must contain unique usernames");
+                }
+                entries.put(username, new Entry(username, password, role));
+            } else {
+                throw new InvalidConfigurationException("Invalid configuration provided: " +  "\"" + line + "\". " +
+                        "Cruise Control API credentials config must follow " +
+                        "HashLoginService's file format: username: password [,rolename ...]");
+            }
+        }
+        return entries;
+    }
+
+    /**
+     * Parses auth data from existing Topic Operator API user secret into map of API user entries.
+     *
+     * @param secret API user secret
+     *
+     * @return Map of API user entries containing user-managed API user credentials
+     */
+    public static Map<String, Entry> generateToManagedApiCredentials(Secret secret) {
+        Map<String, Entry> entries = new HashMap<>();
+        if (secret != null) {
+            if (secret.getData().containsKey(TOPIC_OPERATOR_USERNAME_KEY) && secret.getData().containsKey(TOPIC_OPERATOR_PASSWORD_KEY)) {
+                String username = Util.decodeFromBase64(secret.getData().get(TOPIC_OPERATOR_USERNAME_KEY));
+                String password = Util.decodeFromBase64(secret.getData().get(TOPIC_OPERATOR_PASSWORD_KEY));
+                entries.put(username, new Entry(username, password, Role.ADMIN));
+            }
+        }
+        return entries;
+    }
+
+    /**
+     * Ensures that no illegal roles are used in given API user entries.
+     *
+     * @param entries API credential entries containing username, passwords, and roles.
+     */
+    private static void checkRoles(Map<String, Entry> entries) {
+        for (Entry entry : entries.values()) {
+            String username = entry.username();
+            Role role =  entry.role();
```

Comment:
I do not think you really need to create these local variables. Why don't you just use the `entry.xxx()` directly?